### DISCUSSION TITLE
Cannot build HelloCpp.exe

### DISCUSSION THREAD

**Discussion by awetos** [axmolengine/axmol#1339] [Q&A] [answered]
I'm not sure if I am missing something. I have tried building HelloCpp. 
The project is created with "axmol new -p org.axmol.hellocpp -d "C:\Users\aweto\OneDrive\Desktop\Axmol Projects"-l cpp --portrait HelloCpp" per dev setup. I go into the folder build_x64, open the sln, and then click VS's run, which tries to build it. 

![when building from vs studio](https://github.com/axmolengine/axmol/assets/78708899/0f07bb1b-02ee-48ba-95a6-ced5cbd16eb3)
These are the errors I get when doing that.

I also try the powershell method by entering $axmol build while in the directory.
![building in powershell](https://github.com/axmolengine/axmol/assets/78708899/433f411c-a032-4069-b0c4-fc8a62b72884)

Both have errors and I cannot get HelloCpp to run. I downloaded axmol-2.0.0.

**Comment by halx99** [comment]
try does the latest dev works?

**Comment by halx99** [comment]
maybe caused bu path contains space

**Reply by rh101** [reply]
That is most likely the issue. This comes up when creating the project via cmake (not downloading WebView etc.):

![image](https://github.com/axmolengine/axmol/assets/8603230/91fef6b2-04bf-47a6-88e7-3ef019e6e217)

This was tested with branches 1.0 and dev.

**Comment by halx99** [comment, marked as answer]
> That is most likely the issue. This comes up when creating the project via cmake (not downloading WebView etc.):
> 
> ![image](https://user-images.githubusercontent.com/8603230/269131444-91fef6b2-04bf-47a6-88e7-3ef019e6e217.png)
> 
> This was tested with branches 1.0 and dev.

Fixed by https://github.com/axmolengine/axmol/commit/73d7f9ad268018d5b82e55b0c44b9e83d2ac7d36

**Comment by awetos** [comment]
Thank you for fixing that. I no longer have the webview issue.
It now has 2 errors instead of 5.
![Shader Failed](https://github.com/axmolengine/axmol/assets/78708899/43f52c1a-7d21-4eaf-b9b0-6bb98abc3663)
Now I am missing some sort of shader? 

![SetLocal](https://github.com/axmolengine/axmol/assets/78708899/8eff36fb-f7b8-4b7c-95e4-a290e1584222)
Also, what do the SetLocal errors mean?

**Reply by iAndyHD3** [reply]
it's a script error. 

for temp fix copy the `axslc` folder that is in `build/runtime/` into the folder where the executable is(my path is `build\bin\cpp_tests\Debug`)

**Comment by halx99** [comment]
you can copy the full cmdline fsync.ps1 to terminal the execute, will report more detail error mesaage

**Comment by awetos** [comment]
> it's a script error.
> 
> for temp fix copy the `axslc` folder that is in `build/runtime/` into the folder where the executable is(my path is `build\bin\cpp_tests\Debug`)

This seems to have fixed it, so now the program is running and displaying graphics now 
![blank](https://github.com/axmolengine/axmol/assets/78708899/6c9037be-a9ae-45f8-9704-e7bf3fab7f6b)

I still get the 2 set local errors:
`Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3073	The command "setlocal
"C:\Program Files\PowerShell\7\pwsh.exe" C:/Users/aweto/OneDrive/Desktop/axmol-dev/1k/fsync.ps1 -s C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/build_x64/runtime/axslc -d C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/build_x64/bin/HelloCpp/Debug/axslc -l TRUE
if %errorlevel% neq 0 goto :cmEnd
:cmEnd
endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
:cmErrorLevel
exit /b %1
:cmDone
if %errorlevel% neq 0 goto :VCEnd
:VCEnd" exited with code 1.	SYNC_RESOURCE-HelloCpp-axslc	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	161	

Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3073	The command "setlocal
"C:\Program Files\PowerShell\7\pwsh.exe" C:/Users/aweto/OneDrive/Desktop/axmol-dev/1k/fsync.ps1 -s C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/Content -d C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/build_x64/bin/HelloCpp/Debug/Content -l TRUE
if %errorlevel% neq 0 goto :cmEnd
:cmEnd
endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
:cmErrorLevel
exit /b %1
:cmDone
if %errorlevel% neq 0 goto :VCEnd
:VCEnd" exited with code 1.	SYNC_RESOURCE-HelloCpp	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	161	
`

Also, when I open the HelloCpp.exe from the \build_x64\bin\HelloCpp\Debug I don't get the images anymore and it is just blank. But at least there are no errors. 

![blank from exe](https://github.com/axmolengine/axmol/assets/78708899/b896e3b7-23dd-44b2-aaa3-b2610186757f)

**Comment by halx99** [comment]
i mean run `"C:\Program Files\PowerShell\7\pwsh.exe" C:/Users/aweto/OneDrive/Desktop/axmol-dev/1k/fsync.ps1 -s C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/build_x64/runtime/axslc -d C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/build_x64/bin/HelloCpp/Debug/axslc -l TRUE` in cmd.exe terminal

**Comment by awetos** [comment]
> i mean run `"C:\Program Files\PowerShell\7\pwsh.exe" C:/Users/aweto/OneDrive/Desktop/axmol-dev/1k/fsync.ps1 -s C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/build_x64/runtime/axslc -d C:/Users/aweto/OneDrive/Desktop/HelloCpp/HelloCpp/build_x64/bin/HelloCpp/Debug/axslc -l TRUE` in cmd.exe terminal

Maybe because I already drag and dropped axlsc folder into bin/HelloCpp/debug I first get this error.
![image](https://github.com/axmolengine/axmol/assets/78708899/d3d25447-a057-4916-b07c-44ac1bc51ad0)

Then I deleted that folder, and ran the code again. 
![image](https://github.com/axmolengine/axmol/assets/78708899/955e118a-abb8-40b2-87ca-5bcbcedb6494)

It created a shortcut to the axlsc folder. When I click HelloCpp.exe (Am I clicking the wrong thing?) I still do not get any images though, unless I build and play through VS studio. 

When I play from VS studio, the images show, I still get the setlocal error. 
![image](https://github.com/axmolengine/axmol/assets/78708899/eed54a82-66ce-4a6d-84b9-8c2bdad6ccf8)

**Reply by halx99** [reply]
![image](https://github.com/axmolengine/axmol/assets/6977319/8f5cf2a6-1ef5-4b91-be10-caa12918b968)

**Reply by KorotkiyKamil** [reply]
After a new compilation these folders disappear.

**Reply by KorotkiyKamil** [reply]
![axmol_error_edit_0](https://github.com/axmolengine/axmol/assets/137780670/e210717c-785b-4d91-a664-bb3ec1f8be9d)

**Comment by awetos** [comment]
Thank you, it works now
![image](https://github.com/axmolengine/axmol/assets/78708899/05b28cc9-e070-41eb-b13d-98275663b9f7)
